fiftyone/factory/repos/execution_store.py

```diff
@@ -1,35 +1,30 @@
 """
 Execution store repository.
+
+| Copyright 2017-2024, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
 """
 
-import datetime
-import bson
-from pymongo.collection import Collection
-from fiftyone.operators.store.models import StoreDocument, KeyDocument
+from datetime import datetime
 
+from bson import ObjectId
+from pymongo.collection import Collection
 
-def _where(store_name, key=None, dataset_id=None):
-    query = {"store_name": store_name}
-    if key is not None:
-        query["key"] = key
-    if dataset_id is not None:
-        query["dataset_id"] = dataset_id
-    return query
+from fiftyone.operators.store.models import StoreDocument, KeyDocument
 
 
-class ExecutionStoreRepo:
+class ExecutionStoreRepo(object):
     """Base class for execution store repositories."""
 
     COLLECTION_NAME = "execution_store"
 
-    def __init__(
-        self, collection: Collection, dataset_id: bson.ObjectId = None
-    ):
+    def __init__(self, collection: Collection, dataset_id: ObjectId = None):
         self._collection = collection
         self._dataset_id = dataset_id
 
     def create_store(self, store_name, permissions=None) -> StoreDocument:
-        """Creates a store in the execution store."""
+        """Creates a store associated with the current context."""
```

Comment:
context seems misleading here. Its simpler: if a `dataset_id` is provided, it will associate stores with that dataset, and they will be deleted when the dataset is deleted. Context here seems like an overloaded term.

Reply:
I updated the class docstrings to explain what "context" means: https://github.com/voxel51/fiftyone/pull/5086/commits/07a515faa178c42c92542ca095f3affc129b9611